packages/auto-approve/src/auto-approve.ts

```diff
@@ -0,0 +1,226 @@
+ */
+export function handler(app: Probot) {
+  app.on(
+    ['pull_request.opened', 'pull_request.reopened', 'pull_request.edited'],
+    async (context: Context) => {
+      const pr = context.payload;
+      const owner = pr.pull_request.head.repo.owner.login;
+      const repo = pr.pull_request.head.repo.name;
+      const prNumber = pr.number;
+      let config: Configuration | null;
+
+      // Get auto-approve.yml file if it exists
+      // Reading the config requires access to code permissions, which are not
+      // always available for private repositories.
+      try {
+        config = await context.config<Configuration>(CONFIGURATION_FILE_PATH);
+      } catch (err) {
+        err.message = `Error reading configuration: ${err.message}`;
+        logger.error(err);
+        config = null;
+      }
+
+      const PRFiles = await getChangedFiles(
+        context.octokit,
+        owner,
+        repo,
+        prNumber
+      );
+
+      // If there's no config, check to see if the PR is creating the auto-approve.yml file
+      if (!config) {
+        const prConfig = await getBlobFromPRFiles(
+          context.octokit,
+          owner,
+          repo,
+          PRFiles,
+          `.github/${CONFIGURATION_FILE_PATH}`
+        );
+
+        if (prConfig) {
+          // Attempt to get the CODEOWNERS file if it exists
+          const codeOwnersFile = await getBlobFromPRFiles(
+            context.octokit,
+            owner,
+            repo,
+            PRFiles,
+            '.github/CODEOWNERS'
+          );
+
+          // Decide whether to add a passing or failing status checks
+          await logicForConfigCheck(
+            owner,
+            repo,
+            prConfig,
+            codeOwnersFile,
+            context.octokit,
+            context.payload.pull_request.head.sha
+          );
+        } else {
+          // If config isn't in the PR or the repo, ignore the PR
+          logger.info(
+            `Neither repo nor PR contains an auto-approve.yml file, skipping for ${owner}/${repo}/${prNumber}`
+          );
+        }
+      } else {
+        // If there is a config, first confirm that it matches the guidelines
+        // Then, check to see whether the incoming PR matches the config
+
+        // If config is not valid, this function will submit a failing check, and will not merge the PR
+        const isConfigValid = await logicForConfigCheck(
+          owner,
+          repo,
+          config.toString(),
+          undefined,
+          context.octokit,
+          context.payload.pull_request.head.sha
+        );
+
+        // Check to see whether the incoming PR matches the incoming PR
+        const isPRValid = await checkPRAgainstConfig(
+          config,
+          context.payload,
+          context.octokit
+        );
+
+        // If both PR and config are valid, pull in approving-mechanism to tag and approve PR
+        if (isPRValid && isConfigValid) {
```

Comment:
I would be explicit here;

```
ifPrValid === true && isConfigValid === true
```

Otherwise if we ever forget an `await`, we might start merging all pull requests, since promises will evaluate as true:

```js
> if (Promise.resolve(false)) console.info('hello')
hello
```

vs.,

```
> if (Promise.resolve(false) === true) console.info('hello')
undefined
```

Reply:
Nice catch! Thank you!!